File: LeetCode/1877/cpp/lc1877.cpp
#include <algorithm>
#include <cassert>
#include <vector>

class Solution {
public:
  int minPairSum(std::vector<int>& nums) {
    std::sort(nums.begin(), nums.end());
    int ans = -1;
    for (int b = 0, e = nums.size() - 1; b < e; b++, e--) {
      int pairsum = nums[b] + nums[e];
      if (pairsum > ans) {
        ans = pairsum;
      }
    }
    return ans;
  }
};

int main() {
  Solution s;
  {
    std::vector<int> nums = {3, 5, 2, 3};
    assert(s.minPairSum(nums) == 7);
  }
  {
    std::vector<int> nums = {3, 5, 4, 2, 4, 6};
    assert(s.minPairSum(nums) == 8);
  }
}

File: LeetCode/1877/golang/lc1877.go
package lc1877

import "slices"

// Runtime complexity: O(sort)
// Auxiliary space: O(sort)
// Subjective level: easy
// Solved on: 2026-01-24
func minPairSum(nums []int) int {
	slices.Sort(nums)
	ans := -42 // since all elements of `nums` are guaranteed to be positive,
	// initial ans can be any negative number - it will be overwritten in 1st pass.

	L := len(nums) // len(nums) is guaranteed to be even.
	for i := range L / 2 {
		// after sorting, minimum sum requires pairing lowest and highest numbers together.
		ans = max(ans, nums[i]+nums[L-1-i])
	}
	return ans
}
